hazelcast/test/src/cp_test.cpp

```diff
@@ -519,6 +519,24 @@ TEST_F(basic_latch_test, test_multiple_destroy)
     ASSERT_NO_THROW(cp_structure_->destroy().get());
 }
 
+TEST_F(basic_latch_test, test_try_wait)
+{
+    cp_structure_->try_set_count(1).get();
+
+    ASSERT_FALSE(cp_structure_->try_wait().get());
+
+    auto countdown = boost::async(boost::launch::async, [=]() {
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+        cp_structure_->count_down().get();
+    });
+
+    ASSERT_FALSE(cp_structure_->try_wait().get());
+
+    countdown.get();
+
+    ASSERT_TRUE(cp_structure_->try_wait().get());
+}
+
```

Comment:
Depending on timings generally makes the tests flaky. Isn't the code snippet i stated at below accomplish the same behaviour ?

Also I suggested to change latest assertion to `ASSERT_TRUE_EVENTUALLY` because it is told that `try_wait()` might return false even the internal counter is reached to zero at the documentation.

```suggestion
TEST_F(basic_latch_test, test_try_wait)
{
    cp_structure_->try_set_count(1).get();

    ASSERT_FALSE(cp_structure_->try_wait().get());

    auto countdown = boost::async(boost::launch::async, [=]() {
        cp_structure_->count_down().get();
    });

    countdown.get();

    ASSERT_TRUE_EVENTUALLY(cp_structure_->try_wait().get());
}
```

Reply:
sleep and async part is removed and ASSERT_TRUE_EVENTUALLY is used.

Reply:
Fixed